openeo/rest/multiresult.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from typing import Dict, List, Optional
+
+from openeo import BatchJob
+from openeo.internal.graph_building import FlatGraphableMixin, MultiLeafGraph
+from openeo.rest import OpenEoClientException
+from openeo.rest.connection import Connection, extract_connections
+
+
+class MultiResult(FlatGraphableMixin):
+    """
+    Adapter to create/run batch jobs from process graphs with multiple end/result/leaf nodes.
+
+    Usage example:
+
+    .. code-block:: python
+
+        cube1 = ...
+        cube2 = ...
+        multi_result = MultiResult([cube1, cube2])
+        job = multi_result.create_job()
+
+
+    """
+
+    def __init__(self, leaves: List[FlatGraphableMixin], connection: Optional[Connection] = None):
+        self._multi_leaf_graph = MultiLeafGraph(leaves=leaves)
+        self._connection = self._common_connection(leaves=leaves, connection=connection)
```

Comment:
TODO: fail on finding common connection should be lazy (when user actually triggers need for a connection)

Reply:
won't fix: MultiResult is really connection-centric and should "fail fast" here. if connection-less flattening is desired, `MultiLeafGraph` can be used